pkg/envinfo/url.go

```diff
@@ -15,31 +14,45 @@ import (
 	"github.com/pkg/errors"
 )
 
-// GetPorts returns the ports, returns default if nil
-func (ei *EnvInfo) GetPorts() ([]string, error) {
+//getPorts gets the ports from devfile
+func (ei *EnvInfo) getPorts(container string) ([]string, error) {
 	var portList []string
-
+	portMap := make(map[string]bool)
```

Comment:
I think you can let the definition of portMap at its previous place, as it is not used before.

Reply:
Moved to just before for loop